TokenBucket.php

```diff
@@ -0,0 +1,146 @@
+<?
+
+namespace iFixit\TokenBucket;
+
+use \InvalidArgumentException;
+
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'TokenRate.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'Backend.php';
+
+/**
+ * Implements the token bucket algorithm and stores tokens in storage.
+ *
+ * The rate of token regeneration is specified by the TokenRate class. This
+ * also specifies the maximum number of tokens the bucket can hold.
+ *
+ * The current number of tokens is time sensative so it is not stored as a
+ * member of the class, instead you will have to call getTokens() to calculate
+ * and retrieve the current number of tokens.
+ *
+ * Last consume can possibly be changed by another concurrent request, and
+ * therefore is also not stored in the Object's state and always retrieved from
+ * storage when needed.
+ */
+class TokenBucket {
+   // String to identify this bucket uniquely in storage.
+   private $identifier;
+   // TokenRate is when your parents try to ration your arcade spending.
+   private $rate;
+   // Where the token is to be stored.
+   private $storage;
```

Comment:
It wasn't used, I fixed the name here 70e2a069239835f04359b599ac4763c124935717

This should have been backend, but it's used to inject how the bucket should be stored.